doc/source/exclude_patterns.md

```diff
@@ -0,0 +1,44 @@
+
+The feature prevents given paths and hashes of 'materials' (files before command execution) and 'products' (files after command execution) from being written together with other information (executed command, return value, stdout and stderr) in generating link metadata files. When a given pattern's directory is excluded, all its files and subdirectories are excluded as well.
+
+# Pattern Format
+  - A backslash must be included in front of patterns beginning with a literal hash or exclamation mark. 
+  - The slash is a directory seperator. Separators at beginning or middle (or both) of a pattern are relative to directory level of particular .gitignore file. Separators at the end of the pattern only  match directories. 
+  - Beginning double asterisks match any leading path segments, inner double asterisks match multiple (or zero) inner path segments, and ending double asterisks match any trailing path segments. Single asterisks match everything except a slash. 
+  - Question marks match any one character except a slash.
+
+## Special Patterns
+1. ### Pattern beginning with a slash ###
+  - Match paths on the root directly instead of any descendant paths by making a pattern relative to root. Escape the relative root conversion with an extra forward slash in front so as to match absolute paths. (i.e, "//"). 
+
+2. ### Pattern that is a slash ###
+  - According to 'git check-ignore' (v2.4.1), a single '/' does not match any file
+
+3. ### Patterns without beginning slash ###
+  - A pattern without a beginning slash but contains at least one prepended directory (e.g ``` dir/{pattern} ```) should not match ``` **/dir/{pattern} ``` according to 'git-check-ignore' (v2.4.1.)
+  - A single pattern without a beginning slash will match any descendant path, equivalent to ``` **/{pattern} ```. So, prepend with double-asterisks to make a pattern relative to the root. This also holds for a single pattern with a trailing slash (e.g. ```dir/```). 
+
+4. ### Patterns ending with a slash ###
+  - A pattern ending with a slash ('/' will match all descendant paths if it is a directory but not if it is a regular file, equivalent to ``` {pattern}/** ```. So, set the last segment to double asterisks to include all descendants. 
+
+5. ### Patterns beginning with a literal hash ###
+  - Serves as a comment (neither includes or excludes files). Escape the hash with a back-slash to match a literal hash (i.e., '\#'). 
+
+6. ### Patterns beginning with an exclamation mark ###
+  - Matched patterns will be unmatched. Escape the exclamation mark with a back-slash to match a literal exclamation mark (i.e, '\!').
+
+## Pattern Format Examples
+  - The pattern ``` hello.* ``` matches any file or folder whose name begins with **hello**. If one wants to restrict this only to the directory and not in its subdirectories, one can prepend the pattern with a slash, i.e. ``` /hello.*``` ; the pattern now matches ``` hello.txt, hello.c```  but not ``` a/hello.java ```. 
```

Comment:
```suggestion
  - The pattern `hello.*` matches any file or folder whose name begins with **hello**. If one wants to restrict this only to the directory and not in its subdirectories, one can prepend the pattern with a slash, i.e. `/hello.*`; the pattern now matches `hello.txt, hello.c` but not `a/hello.java`. 
```

Reply:
There are some spacing inconsistencies here and elsewhere. Also, a single backtick is enough to enclose code in-line.

Reply:
Feel free to mark a review comment thread as resolved when done. :)